tests/test_bootstrap_archive.py

```
@@ -0,0 +1,436 @@
+        with self.installed_fixture() as (directory, _):
+            layout = [locked_member()]
+            with open_validated_executable(directory, layout, "tool") as source:
+                with mock.patch("bootstrap_archive.os.memfd_create", capture_memfd):
+                    with mock.patch(
+                        "bootstrap_archive.fcntl.fcntl", side_effect=OSError("fail")
+                    ):
+                        with self.assertRaises(BootstrapError):
+                            with open_sealed_executable(source, layout[0]):
+                                self.fail("unsealed executable was yielded")
+        self.assertEqual(len(captured), 1)
+        with self.assertRaises(OSError):
```

Ruff (0.16.1)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)